lib/que/job.rb

```diff
@@ -167,49 +176,43 @@ def _bulk_enqueue_insert(args_and_kwargs_array, job_options: {}, notify:)
-          run_at:   job_options[:run_at]   || resolve_que_setting(:run_at),
-          args_and_kwargs_array: args_and_kwargs_array,
-          data:     job_options[:tags] ? { tags: job_options[:tags] } : {},
-          job_class: \
-            job_options[:job_class] || name ||
-              raise(Error, "Can't enqueue an anonymous subclass of Que::Job"),
-        }
-
-        if attrs[:run_at].nil? && resolve_que_setting(:run_synchronously)
-          args_and_kwargs_array = Que.deserialize_json(Que.serialize_json(attrs.delete(:args_and_kwargs_array)))
-          args_and_kwargs_array.map do |args_and_kwargs|
-            _run_attrs(
-              attrs.merge(
-                args: args_and_kwargs.fetch(:args),
-                kwargs: args_and_kwargs.fetch(:kwargs),
-              ),
+        jobs_attrs = jobs_attrs.map do |attrs|
+          klass = attrs[:klass] || self
+
+          attrs = {
+            queue:     attrs[:queue]     || job_options[:queue]     || klass.resolve_que_setting(:queue) || Que.default_queue,
+            priority:  attrs[:priority]  || job_options[:priority]  || klass.resolve_que_setting(:priority),
+            run_at:    attrs[:run_at]    || job_options[:run_at]    || klass.resolve_que_setting(:run_at),
+            job_class: attrs[:job_class] || job_options[:job_class] || klass.name,
+            args:      attrs[:args]      || [],
+            kwargs:    attrs[:kwargs]    || {},
+            data:      attrs[:data]      || (job_options[:tags] ? { tags: job_options[:tags] } : {}),
+            klass:     klass
+          }
+
+          if attrs[:run_at].nil? && klass.resolve_que_setting(:run_synchronously)
+            klass._run_attrs(
+              attrs.except(:klass).merge(
+                args: Que.deserialize_json(Que.serialize_json(attrs[:args])),
+                kwargs: Que.deserialize_json(Que.serialize_json(attrs[:kwargs])),
+                data: Que.deserialize_json(Que.serialize_json(attrs[:data])),
+              )
             )
+            nil
+          else
+            attrs
           end
-        else
-          attrs.merge!(
-            args_and_kwargs_array: Que.serialize_json(attrs[:args_and_kwargs_array]),
-            data: Que.serialize_json(attrs[:data]),
-          )
-          values_array =
-            Que.transaction do
-              Que.execute('SET LOCAL que.skip_notify TO true') unless notify
-              Que.execute(
-                :bulk_insert_jobs,
-                attrs.values_at(:queue, :priority, :run_at, :job_class, :args_and_kwargs_array, :data),
-              )
-            end
-          values_array.map(&method(:new))
-        end
+        end.compact
+
+        values_array =
+          Que.transaction do
+            Que.execute('SET LOCAL que.skip_notify TO true') unless notify
+            Que.execute(
+              :bulk_insert_jobs,
+              [Que.serialize_json(jobs_attrs.map { |attrs| attrs.except(:klass) })]
```

Comment:
Ruby 2.7 annoyingly does not have `Hash#except`. iirc, I've used `Hash#reject` for this previously.

Reply:
Yeah, I noticed this broke CI for Ruby 2.7. I'll fix this.